src/modules/discovery/hosts.py

```diff
@@ -38,44 +41,68 @@ def get_service_account_file(self, file):
-        self.predefined_hosts = predefined_hosts
+    pass
 
-class HostDiscoveryHelpers:
-    @staticmethod
-    def get_cloud(host):
+# for comparing prefixes
+class InterfaceTypes(Enum):
+    LOCALHOST = "127"
+
+class HostDiscoveryUtils:
+    """ Static class containes util functions for Host discovery processes """
+    @classmethod
+    def get_cloud(cls, host):
+        """ Returns cloud for a given ip address, defaults to the external ip"""
+        cloud = ""
         try:
-            logging.debug("Checking whether the cluster is deployed on azure's cloud")
-            # azurespeed.com provide their API via HTTP only; the service can be queried with 
-            # HTTPS, but doesn't show a proper certificate. Since no encryption is worse then
-            # any encryption, we go with the verify=false option for the time being. At least
-            # this prevents leaking internal IP addresses to passive eavesdropping.
-            # TODO: find a more secure service to detect cloud IPs
-            metadata = requests.get("https://www.azurespeed.com/api/region?ipOrUrl={ip}".format(ip=host), verify=False).text
+            if host:
+                logging.debug("Checking if {} is deployed on a cloud".format(host))
+                # azurespeed.com provide their API via HTTP only; the service can be queried with 
+                # HTTPS, but doesn't show a proper certificate. Since no encryption is worse then
+                # any encryption, we go with the verify=false option for the time being. At least
+                # this prevents leaking internal IP addresses to passive eavesdropping.
+                # TODO: find a more secure service to detect cloud IPs
+                metadata = requests.get("https://www.azurespeed.com/api/region?ipOrUrl={ip}".format(ip=host), verify=False).text
+                if "cloud" in metadata:
+                    cloud = json.loads(metadata)["cloud"]
         except requests.ConnectionError as e:
             logging.info("- unable to check cloud: {0}".format(e))
-            return
-        if "cloud" in metadata:
-            return json.loads(metadata)["cloud"]
 
-    # generator, generating a subnet by given a cidr
+        return CloudTypes.get_enum(cloud)
+
+    @staticmethod 
+    def get_default_gateway():
+        return gateways()['default'][AF_INET][0]
+
+    @staticmethod
+    def get_external_ip():
+        external_ip = None
+        try:
+            logging.debug("HostDiscovery hunter attempting to get external IP address")
+            external_ip = requests.get("http://canhazip.com").text # getting external ip, to determine if cloud cluster
+        except requests.ConnectionError as e:
+            logging.debug("unable to determine external IP address: {0}".format(e))
+        return external_ip
+
+    # generator, generating ip addresses from a given cidr
     @staticmethod
     def generate_subnet(ip, sn="24"):
```

Comment:
I'm not totally sure about this but I wonder if the places that call generate_subnet should just call IPNetwork directly. I think this would save generating a list, and then turning it back into a generator again with yield, right?

Reply:
I think you're right! how about just returning an IPNetwork("{}/{}".format(ip, sn)) from this function?
So that we will keep the usage in the utils, and keep the default subnet to 24?